lib/_stream_readable.js

```diff
@@ -543,10 +543,16 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
   // is attached before any userland ones.  NEVER DO THIS.
   if (!dest._events || !dest._events.error)
     dest.on('error', onerror);
-  else if (Array.isArray(dest._events.error))
-    dest._events.error.unshift(onerror);
-  else
-    dest._events.error = [onerror, dest._events.error];
+  else {
+    var existingHandler = dest._events.error;
+    if (Array.isArray(existingHandler))
+      existingHandler.unshift(onerror);
+    else {
+      var handlers = [onerror, existingHandler];
+      handlers.onceCount = (existingHandler.once ? 1 : 0);
+      dest._events.error = handlers;
+    }
+  }
```

Comment:
@chrisdickinson If we simultaneously merge this same change into iojs/readable-stream everything should still work for both node.js and io.js right?